load.php

```diff
@@ -134,7 +134,25 @@ function( $module_settings ) {
 	 */
 	$modules = apply_filters( 'perflab_active_modules', $modules );
 
-	return $modules;
+	$active_modules = array();
+	foreach ( $modules as $module ) {
+
+		// Do not load module if it marge in WordPress.
+		$can_load_module = perflab_can_load_module( $module );
+		if ( ! $can_load_module ) {
+			continue;
+		}
+
+		// Do not load module if it no longer exists.
+		$module_file = plugin_dir_path( __FILE__ ) . 'modules/' . $module . '/load.php';
+		if ( ! file_exists( $module_file ) ) {
+			continue;
+		}
+
+		$active_modules[] = $module;
+	}
+
+	return $active_modules;
```

Comment:
It is worth discussing, but I think it would be more appropriate to put all of this logic into the `perflab_load_active_modules()` function. Which modules are technically active isn't affected by which modules can be loaded, and I think it makes more sense to handle the whole logic related to loading a module in that `perflab_load_active_modules()` function.

This keeps the two functions more clearly separated. By adding all this logic here, we basically change this function into a "get active and loadable modules" function, but it is only `perflab_get_active_modules()`.

I'd say we should either move all these checks (back) into `perflab_load_active_modules()`, or alternatively we could introduce a completely new function, e.g. `perflab_get_active_and_valid_modules()` for it. (That name would make sense for parity with the similar core function [`wp_get_active_and_valid_plugins()`](https://developer.wordpress.org/reference/functions/wp_get_active_and_valid_plugins/).)

Reply:
Thanks for the review and explanation, Felix.

`perflab_get_active_modules()` called in two functions `perflab_get_generator_content()` and `perflab_load_active_modules()` in which we get active module lists, so I update the `perflab_get_active_modules()` function code so we don't need to add duplicate code for both functions.

If we add a new function perflab_load_active_modules(), still we have to use the if statement in both functions to determine whether a module is active or not?

    if ( ! perflab_load_active_modules( $active_modules ) ) {
        return;
    }

Or should we `perflab_get_active_modules()` to `rename perflab_get_active_modules()` with a document that contains both logic in a single function?